pyincore/utils/popdisloutputprocess.py

```diff
@@ -46,31 +46,31 @@ def __init__(self, pop_disl_result, pop_disl_result_path=None,
                 pd_result_flag = pd_result[(pd_result["guid"].notnull()) &
                                            (pd_result["numprec"].notnull()) &
                                            (pd_result["plcname10"] == filter_name)]
-                # only keep guid, place and dislocated
-                pd_result_shp = pd_result[(pd_result["dislocated"]) &
-                                          (pd_result["guid"].notnull()) &
+                # only keep guid, place
+                pd_result_shp = pd_result[(pd_result["guid"].notnull()) &
                                           (pd_result["numprec"].notnull()) &
                                           (pd_result["plcname10"] == filter_name)]
             else:
                 pd_result_flag = pd_result[(pd_result["guid"].notnull()) &
                                            (pd_result["numprec"].notnull())]
                 # only keep guid and dislocated
-                pd_result_shp = pd_result[(pd_result["dislocated"]) &
-                                          (pd_result["guid"].notnull()) &
+                pd_result_shp = pd_result[(pd_result["guid"].notnull()) &
                                           (pd_result["numprec"].notnull())]
         else:
             if filter_name:
                 pd_result_flag = pd_result[(pd_result["numprec"].notnull()) &
                                            (pd_result["plcname10"] == filter_name)]
-                # only keep guid, place and dislocated
-                pd_result_shp = pd_result[(pd_result["dislocated"]) &
-                                          (pd_result["numprec"].notnull()) &
+                # only keep guid, place
+                pd_result_shp = pd_result[(pd_result["numprec"].notnull()) &
                                           (pd_result["plcname10"] == filter_name)]
             else:
                 pd_result_flag = pd_result[(pd_result["numprec"].notnull())]
-                # only keep guid and dislocated
-                pd_result_shp = pd_result[(pd_result["dislocated"]) &
-                                          (pd_result["numprec"].notnull())]
+                # only keep guid
+                pd_result_shp = pd_result[(pd_result["numprec"].notnull())]
+
+        # Update "numprec" to 0 where "dislocated" is False
+        pd_result_shp.loc[pd_result_shp["dislocated"], "numprec"] = 0
```

Comment:
Correct me if I'm wrong @longshuicy, but I don't think this is necessary. I think numprec is just the number of people in the household so the fact they aren't dislocated doesn't need to modify anything. I think this step can be removed, right?

Reply:
Now I understand, the numprec value is getting populated with the value coming from HUA if the household dislocates. Since we are including all households, any that aren't dislocated should have a numprec of 0. Thanks @longshuicy

Reply:
@ylyangtw maybe just to be safe, could we just rename this field to "numprec_dislocated" so the original numprec still left untouched. Do you think it's worth it?

Reply:
If we rename it to "numprec_dislocated", the non-dislocated items under original numprec need to be False/0/Null. These two columns will be duplicated I think. So probably we don't need to keep the original.

Reply:
Let's not do that then. I will approve your PR as it is :-)